include/aws/common/task_scheduler.h

```diff
@@ -68,26 +75,23 @@ AWS_COMMON_API
 void aws_task_scheduler_clean_up(struct aws_task_scheduler *scheduler);
 
 /**
- * Gets the next task that is ready for execution. Tasks in the future will not be returned,
- * even if they are the highest priority. next_run_time, if it is not null, and there are any tasks scheduled
- * will be set to the timestamp for the highest priority task. This is a useful hint for setting timeouts on
- * event loops, or thread sleeps. If no tasks are scheduled, this value will be set to 0.
- * If no tasks are ready for execution AWS_ERROR_TASK_SCHEDULER_NO_READY_TASKS error will be raised.
- *
- * If no tasks are scheduled AWS_ERROR_TASK_SCHEDULER_NO_TASKS error will be raised.
- * task is copied.
+ * Returns whether the scheduler has any scheduled tasks.
+ * next_task_time (optional) will be set to time of the next task, note that 0 will be set if tasks were
+ * added via aws_task_scheduler_schedule_now() and UINT64_MAX will be set if no tasks are scheduled at all.
```

Comment:
I like it when out-params are always set to something. I thought UINT64_MAX was a good value to use, because it basically means "the next task is infinity away"